i18n/en/docusaurus-plugin-content-docs/current/guides/tech/with-nextjs.mdx

```diff
@@ -1,72 +1,73 @@
 
-NextJS suggests using the `pages` folder to define application routes. NextJS expects files in the `pages` folder to match URLs.
-This routing mechanism **does not correspond** to the FSD concept, since it is not possible to maintain a flat slice structure in such a routing mechanism.
+You should also add a `pages` folder to the project root because App Router is compatible with Pages Router.
+The best way is to put an empty `.gitkeep` or `README.md` file inside describing the need to save this folder.
 
-### Moving the `pages` folder of NextJS to the root folder of the project (recommended)
+```sh
+├── app              # App folder (Next.js)
+├── pages            # Legacy pages folder (Next.js)
+│   └── .gitkeep     # To maintain compatibility with Pages Router
+└── src
+    ├── app
+    ├── pages
+    ├── widgets
+    ├── features
+    ├── entities
+    └── shared
+```
 
-The approach is to move the `pages` NextJS folder to the root folder of the project and import the FSD pages into the `pages` NextJS folder. This saves
-the FSD project structure inside the `src` folder.
+### Example of re-exporting a page from `src/pages/example` to `app/example/page.tsx`
 
-```sh
-├── pages              # NextJS pages folder
-├── src
-│   ├── app
-│   ├── entities
-│   ├── features
-│   ├── pages          # FSD pages folder
-│   ├── shared
-│   ├── widgets
+```tsx title="app/example/page.tsx"
+export { Example as default, metadata } from '@/pages/example';
 ```
 
-### Renaming the `pages` layer within the FSD structure
+### Middleware
+If you use middleware in your project, it must be located in the project root alongside the `app` and `pages` folders.
 
-Another way to solve the problem is to rename the `pages` layer in the FSD structure to avoid conflicts with the NextJS `pages` folder.
-You can rename the `pages` layer in FSD to `views`.
-In that way, the structure of the project in the `src` folder is preserved without contradiction with the requirements of NextJS.
+### Instrumentation
+The `instrumentation.ts|js` file allows you to monitor the performance and behavior of your application. If you use it, it must be located in the project root, similar to `middleware.ts|js`.
 
-```sh
-├── app
-├── entities
-├── features
-├── pages              # NextJS pages folder
-├── views              # Renamed FSD pages folder
-├── shared
-├── widgets
+### Route Handlers
+It is suggested to use a new `routes` segment in the `app` or `pages` layer to work with Route Handlers.
+
+```tsx title="app/api/get-example/route.ts"
+export { getExample as GET } from '@/pages/example';
 ```
 
-Keep in mind that it's highly recommended to document this rename prominently in your project's README or internal documentation. This rename is a part of your ["project knowledge"][project-knowledge].
+### Additional recommendations
+- Use the `queries` segment in the `shared` layer to describe database queries and their further use in higher layers.
```

Comment:
issue: I think that `queries` here describes what the code is, not what the code is for (see https://github.com/feature-sliced/steiger/blob/master/packages/steiger-plugin-fsd/src/segments-by-purpose/README.md)

suggestion: I prefer to keep the logic of server actions and data loaders in the `api` segment of pages that need it, but let's discuss this in more detail too